packages/health-check/src/checks/critical.js

```diff
@@ -7,53 +7,86 @@ const { calculateElapsedTime, getResponseContent } = require("../utils");
   const time = process.hrtime();
   const form = new FormData();
-  const data = Buffer.from(new Date()); // current date to ensure data uniqueness
+  const payload = Buffer.from(new Date()); // current date to ensure data uniqueness
+  const data = { up: false };
 
-  form.append("file", data, { filename: "time.txt", contentType: "text/plain" });
-  let statusCode, errorResponseContent;
+  form.append("file", payload, { filename: "time.txt", contentType: "text/plain" });
 
   try {
-    const response = await got.post(`${process.env.PORTAL_URL}/skynet/skyfile`, { body: form });
+    const response = await got.post(`${process.env.SKYNET_PORTAL_API}/skynet/skyfile`, { body: form });
 
-    statusCode = response.statusCode;
+    data.statusCode = response.statusCode;
+    data.up = true;
+    data.ip = response.ip;
   } catch (error) {
-    statusCode = error?.response?.statusCode || error.statusCode || error.status;
-    errorResponseContent = getResponseContent(error?.response);
+    data.statusCode = error?.response?.statusCode || error.statusCode || error.status;
+    data.errorMessage = error.message;
+    data.errorResponseContent = getResponseContent(error.response);
+    data.ip = error?.response?.ip ?? null;
   }
 
   done({
     name: "upload_file",
-    up: statusCode === StatusCodes.OK,
-    statusCode,
-    errorResponseContent,
     time: calculateElapsedTime(time),
+    ...data,
   });
 }
 
 // downloadCheck returns the result of downloading the hard coded link
 async function downloadCheck(done) {
   const time = process.hrtime();
   const skylink = "AACogzrAimYPG42tDOKhS3lXZD8YvlF8Q8R17afe95iV2Q";
-  let statusCode, errorMessage, errorResponseContent;
+  const data = { up: false };
 
   try {
-    const response = await got(`${process.env.PORTAL_URL}/${skylink}?nocache=true`);
+    const response = await got(`${process.env.SKYNET_PORTAL_API}/${skylink}?nocache=true`);
 
-    statusCode = response.statusCode;
+    data.statusCode = response.statusCode;
+    data.up = true;
+    data.ip = response.ip;
   } catch (error) {
-    statusCode = error?.response?.statusCode || error.statusCode || error.status;
-    errorMessage = error.message;
-    errorResponseContent = getResponseContent(error.response);
+    data.statusCode = error?.response?.statusCode || error.statusCode || error.status;
+    data.errorMessage = error.message;
+    data.errorResponseContent = getResponseContent(error.response);
+    data.ip = error?.response?.ip ?? null;
   }
 
   done({
     name: "download_file",
-    up: statusCode === StatusCodes.OK,
-    statusCode,
-    errorMessage,
-    errorResponseContent,
     time: calculateElapsedTime(time),
+    ...data,
   });
 }
 
-module.exports = [uploadCheck, downloadCheck];
+async function accountHealthCheck(done) {
+  const time = process.hrtime();
+  const data = { up: false };
+
+  try {
+    const response = await got(`${process.env.SKYNET_DASHBOARD_URL}/health`, { responseType: "json" });
```

Comment:
Same comment as above. Maybe we can use the value of the env var in and bind that to the current IP? That would remove the possibility of desyncing the two. But while we're using one hardcoded value (Dockerfile) and another configurable one we will be running that risk.

Reply:
`SKYNET_DASHBOARD_URL` contains `https://` prefix and it wouldn't work if we assign the ip to address with https:// prefix - tried it. We would need `SKYNET_DASHBOARD_URL_NO_HTTPS` :) Let's follow up in a env cleanup branch once there is a bit more time.

Reply:
1. I agree that `.env` is due for a cleanup. I would advise to do that when we stabilise the stack a little more.
2. We could trim the prefix. That should solve the issue for both `SKYNET_DASHBOARD_URL` and `SKYNET_PORTAL_API`. I would prefer to resolve this now rather than later but I won't hold the PR on it and I'll trust your judgement.